### DISCUSSION TITLE
How do I resolve "ms-resource:TerminalApp/ContextMenu/AppStoreNamePre"

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#12416] [Q&A] [answered]
I was using a built-in TCC variable function (@xmlxpath[]) to dig Properties/DisplayName out of AppxManifest.xml.  With the latest versions, 1.12 stable and 1.13 preview, I'm just getting "ms-resource:TerminalApp/ContextMenu/AppStoreNamePre".  How do I turn that into a display name?  Thanks!  ... Vince

**Comment by DHowett** [comment, marked as answer]
So, this is a localized resource specifier. I think you need to use the [modern resource APIs](https://docs.microsoft.com/en-us/windows/uwp/app-resources/) to resolve it.

For what it's worth, the correct way to handle app packages is to use the [ApplicationModel APIs](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel?view=winrt-22000), not to grovel through XML manually.

**Reply by vefatica** [reply]
Thanks, @DHowett.  I figured it'd be something like that ... overkill for my just needing a string for an enhanced VER command when running in WT.  I can do that in other ways.

**Reply by DHowett** [reply]
I agree that it is unwieldy :(

The powers that be informed us that our app name needs to be localized, unfortunately, and this broke both your TCC macro and PowerToys Run. 😬
